### PR TITLE
Backport Android updateTransform changes

### PR DESCRIPTION
**Pull request by thebrianchen**
After porting update transforms to [Android](https://github.com/firebase/firebase-android-sdk/pull/2262), I found some changes that needed to be ported back. I also replaced usages of TransformMutation.

### CODE REVIEW

File: packages/firestore/src/model/mutation.ts [hunk 1]
```diff
@@ -191,28 +191,22 @@ export function preconditionIsValidForDocument(
...
  * locally, we return an `UnknownDocument` and rely on Watch to send us the
  * updated version.
  *
- * Note that TransformMutations don't create Documents (in the case of being
- * applied to a NoDocument), even though they would on the backend. This is
- * because the client always combines the TransformMutation with a SetMutation
- * or PatchMutation and we only want to apply the transform if the prior
- * mutation resulted in a Document (always true for a SetMutation, but not
- * necessarily for a PatchMutation).
+ * Field transforms are used only with Patcdh and Set Mutations. We use the
```
Comment (wilhuff) [inline packages/firestore/src/model/mutation.ts:207]:
```suggestion
 * Field transforms are used only with Patch and Set Mutations. We use the
```

File: packages/firestore/src/model/mutation.ts [hunk 2]
```diff
@@ -689,14 +675,6 @@ function localTransformResults(
       previousValue = maybeDoc.field(fieldTransform.field);
     }
 
-    if (previousValue === null && baseDoc instanceof Document) {
```
Comment (wilhuff) [inline packages/firestore/src/model/mutation.ts:692]: Why is this logic no longer needed? This seems pretty critical. Or is it handled elsewhere now?

Comment (thebrianchen) [reply packages/firestore/src/model/mutation.ts:692]: My understanding is that this if-statement checks for the edge case where a PatchMutation clears the values before the TransformMutation can be applied. With the addition of updateTransforms, field transformation are encoded on the Patch or Set Mutation, rather than in a separate TransformMutation. This means that we won't bump into this edge case anymore since the update and fieldTransform are on the same mutation.

I've also removed the now obsolete server timestamp test that tested this edge case.

**Review by wilhuff — APPROVED** [review]
LGTM
